common/containers/QrSignerModal/index.tsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import translate, { translateRaw } from 'translations';
+import EthTx from 'ethereumjs-tx';
+import QrSigner from '@parity/qr-signer';
+import { AppState } from 'reducers';
+import Modal, { IButton } from 'components/ui/Modal';
+import './index.scss';
+
+interface State {
+  scan: boolean;
+}
+
+interface PropsClosed {
+  isOpen: false;
+}
+
+interface PropsOpen {
+  isOpen: true;
+  from: string;
+  tx: EthTx;
+  onSignature(signature: string): void;
+  onCancel(): void;
+}
+
+type Props = PropsClosed | PropsOpen;
+
+class QrSignerModal extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      scan: false
+    };
+  }
+
+  public render() {
+    if (!this.props.isOpen) {
+      return null;
+    }
+
+    const { scan } = this.state;
+    const { tx, from } = this.props;
+
+    // Poor man's serialize without signature.
+    // All those values are later overriden by actual signature
+    // values in: wallets/non-deterministic/parity.ts
+    tx.v = Buffer.from([tx._chainId]);
+    tx.r = Buffer.from([0]);
+    tx.s = Buffer.from([0]);
```

Comment:
Looks like you're mutating the `tx` props object here. If you want to alter something passed in as props, you'll need to copy it and save it in state. Something like...

```.tsx
public componentDidMount() {
  if (this.props.tx) {
    this.makeStateTx(this.props);
  }
}

public componentWillReceiveProps(nextProps: Props) {
  if (nextProps.tx && this.props.tx !== nextProps.tx) {
    this.makeStateTx(nextProps);
  }
}

private makeStateTx(props: Props) {
  const tx = { ...props.tx };
  tx.v = Buffer.from([tx._chainId]);
  tx.r = Buffer.from([0]);
  tx.s = Buffer.from([0]);
  this.setState({ tx });
}
```

Reply:
The values aren't read anywhere down the line, so it shouldn't really matter. Ideally there would be a way to serialize a transaction to RLP in ethereumjs without including the signature (`tx.serialize(false)` to match `tx.hash(false)`), but that's not the option. I can make a helper function for it here, and make a PR to ethereumjs.

Reply:
The problem is it makes for non-deterministic redux state, makes it so there's no way to determine if the tx object has changed (without checking deep equality) and [in general is just advised against in React](https://reactjs.org/docs/components-and-props.html#props-are-read-only).

If you want to avoid the state boilerplate, it could at least do
```
const tx = { ...this.props.tx };
```
in `render`.